src/QMCWaveFunctions/Jastrow/J1OrbitalSoA.h

```diff
@@ -140,8 +172,8 @@ struct J1OrbitalSoA : public WaveFunctionComponent
       const auto& displ = d_ie.getDisplRow(iel);
       for (int iat = 0; iat < Nions; iat++)
       {
-        int gid    = Ions.GroupID[iat];
-        auto* func = F[gid];
+        int gid    = Ions.getGroupID(iat);
+        auto* func = J1Unique[gid].get();
         if (func != nullptr)
```

Comment:
How can a I get a func that is a null pointer?  Only if I didn't addfunc for that group so I get no it calls get on an out of bounds J1Unique so no. It could happen if J1Unique[gid] was added as a nullptr which seems like nonsense as well. If for some reason an ion type had no jastrow function a noOpFunction or Optional seems a safer way to handle this.

Reply:
To me, if an ion specie doesn't have a Jastrow, the functor should be nullptr. Adding noOpFunction is just adding source of trouble.

Reply:
I agree to move to `std::optional` in the future, but also that would have to wait until #2699 is required.